src/transaction.js

```diff
@@ -52,93 +66,109 @@ function convertAmountToMinimumUnit(amount_str) {
     return amount;
 }
 
-function transferAmount(amount_data) {
-    let byte_arr = Convert.bigNumberToByteArray(amount_data);
-    let type_arr = new Array(1);
-    type_arr[0] = Constants.AMOUNT_TYPE;
-    let data_arr = type_arr.concat(byte_arr);
-    return data_arr;
-}
-
-function transferShortTxt(description) {
-    let byte_arr = Convert.stringToByteArray(description);
-
-    let type_arr = new Array(1);
-    type_arr[0] = Constants.SHORTTEXT_TYPE ;
-    let length = byte_arr.length;
-    let length_arr = Convert.shortToByteArray(length);
-    return type_arr.concat(length_arr.concat(byte_arr));
-}
-
 function processAttachment(description) {
     return Base58.encode(Convert.stringToByteArray(description));
 }
 
-function transferAccount(account) {
-    let account_arr = Base58.decode(account);
-    let type_arr = [Constants.ACCOUNT_ADDR_TYPE];
-    type_arr = type_arr.concat(Array.from(account_arr));
-    return type_arr;
+function data_bytes_gen(data, data_type) {
+    let data_bytes;
+    switch (data_type) {
+        case Constants.AMOUNT_TYPE:
+            data_bytes = Convert.bigNumberToByteArray(data);
+            break;
+        case Constants.SHORTTEXT_TYPE:
+            let byte_arr = Convert.stringToByteArray(data);
+            let length = byte_arr.length;
+            let length_arr = Convert.shortToByteArray(length);
+            data_bytes = length_arr.concat(byte_arr)
+            break;
+        case Constants.TOKEN_ID_TYPE:
+            let token_id_arr = Base58.decode(data);
+            data_bytes = Array.from(token_id_arr);
+            break;
+        case Constants.ACCOUNT_ADDR_TYPE:
+            let account_arr = Base58.decode(data);
+            data_bytes = Array.from(account_arr);
+            break;
+        case Constants.CONTRACT_ACCOUNT_TYPE:
+            let contract_account_arr = Base58.decode(data);
+            data_bytes = Array.from(contract_account_arr);
+    }
+    return [data_type].concat(data_bytes);
 }
 
-
-// Data process function for registering contract and executing contract(P.S.:name will be changed to 'processTokenContractData' or support all kinds of Contract Data within the function in the future)
-function processContractData(init_data) {
-    if(!init_data.hasOwnProperty('amount')) {
-        throw new Error("There is no field 'amount' in initData");
-    }
-    if(!init_data.hasOwnProperty('unity')) {
-        throw new Error("There is no field 'unity' in initData");
-    }
-    if(!init_data.hasOwnProperty('token_description')) {
-        throw new Error("There is no field 'token_description' in initData");
+// Data process function for registering contract
+function processContractData(init_data, contract_type) {
```

Comment:
Remove `contract_type` here. And `init_data` should be DataEntry Array type 
(like this structure. "name" can be omitted)

```
[
   {
      "name":"amount",
      "type":"Constants.AMOUNT_TYPE",
      "value":10000
   },
   {
      "name":"recipient",
      "type":"Constants.ACCOUNT_ADDR_TYPE",
      "value":"ARcxgg...DFG"
   }
]
```